plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorPropertyTester.java

```diff
@@ -65,6 +68,13 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
             case PROP_CAN_EXECUTE:
                 // Do not check hasActiveQuery - sometimes jface don't update action enablement after cursor change/typing
                 return true;/* && (!"statement".equals(expectedValue) || editor.hasActiveQuery())*/
+            case PROP_CAN_EXECUTE_NATIVELY:
+                if (editor.getDataSource() == null) {
+                    return false;
+                }
+                SQLNativeExecutorDescriptor executorDescriptor = SQLNativeExecutorRegistry.getInstance()
+                    .getExecutorDescriptor(editor.getDataSource());
+                return executorDescriptor != null && executorDescriptor.getNativeExecutor() != null;
```

Comment:
You're using local variables inside a switch case. Please wrap it with braces to avoid polluting the scope:

```suggestion
            case PROP_CAN_EXECUTE_NATIVELY: {
                if (editor.getDataSource() == null) {
                    return false;
                }
                SQLNativeExecutorDescriptor executorDescriptor = SQLNativeExecutorRegistry.getInstance()
                    .getExecutorDescriptor(editor.getDataSource());
                return executorDescriptor != null && executorDescriptor.getNativeExecutor() != null;
            }
```